src/distrobox/mod.rs

```diff
@@ -18,7 +22,110 @@ mod desktop_file;
+    cat
+  else
+    printf '%s' "$1"
+  fi | od -vt x1 -A n | tr -d '[[:space:]]'
+}
+EOF
+)"
+
+eval "$base16FunctionDefinition"
+
+dumpDesktopFiles() {
+  if ! [ -d "$1" ]; then
+    return
+  fi
+
+  find "$1" -name '*.desktop' -not -exec grep -q '^[[:space:]]*NoDisplay[[:space:]]*=[[:space:]]*true[[:space:]]*$' '{}' \; -exec sh -c "$(set +o);$base16FunctionDefinition;"'printf '\''"%s"="%s"\n'\'' "$(base16 "$1")" "$(base16 <"$1")"' - '{}' \;
+}
+
+printf 'home_dir="%s"\n' "$(base16 "$HOME")"
+
+printf '[system]\n'
+dumpDesktopFiles /usr/share/applications
+
+printf '[user]\n'
+dumpDesktopFiles "$HOME/.local/share/applications"
+"#;
+
+#[derive(Deserialize, Debug)]
+struct DesktopFiles {
+    #[serde(deserialize_with = "DesktopFiles::deserialize_path")]
+    home_dir: PathBuf,
+    #[serde(deserialize_with = "DesktopFiles::deserialize_desktop_files")]
+    system: BTreeMap<PathBuf, String>,
+    #[serde(deserialize_with = "DesktopFiles::deserialize_desktop_files")]
+    user: BTreeMap<PathBuf, String>,
+}
+
+impl DesktopFiles {
+    fn decode_hex<E: serde::de::Error>(hex_str: &str) -> Result<Vec<u8>, E> {
+        if hex_str.len() % 2 != 0 {
+            return Err(E::invalid_length(
+                hex_str.len(),
+                &"hex string to have an even lenght",
+            ));
+        }
+
+        (0..hex_str.len())
+            .step_by(2)
+            .map(|i| u8::from_str_radix(&hex_str[i..=i + 1], 16))
+            .collect::<Result<_, _>>()
+            .map_err(|e| {
+                E::custom(format_args!(
+                    "hex string contains non hex characters: {e:?}"
+                ))
+            })
+    }
+
+    fn decode_utf8_from_hex<E: serde::de::Error>(hex_str: &str) -> Result<String, E> {
+        String::from_utf8(Self::decode_hex(hex_str)?).map_err(|e| {
+            E::custom(format_args!(
+                "decoded hex string does not represent valid UTF-8: {e:?}"
+            ))
+        })
+    }
+
+    fn decode_path_from_hex<E: serde::de::Error>(hex_str: &str) -> Result<PathBuf, E> {
+        Ok(PathBuf::from(OsString::from_vec(Self::decode_hex(
+            hex_str,
+        )?)))
+    }
+
+    fn deserialize_path<'de, D: Deserializer<'de>>(deserializer: D) -> Result<PathBuf, D::Error> {
+        Self::decode_path_from_hex(&String::deserialize(deserializer)?)
+    }
+
+    fn deserialize_desktop_files<'de, D: Deserializer<'de>>(
+        deserializer: D,
+    ) -> Result<BTreeMap<PathBuf, String>, D::Error> {
+        BTreeMap::<String, String>::deserialize(deserializer)?
+            .into_iter()
+            .map(|(path, content)| {
+                Ok((
+                    Self::decode_path_from_hex(&path)?,
+                    Self::decode_utf8_from_hex(&content)?,
+                ))
+            })
+            .collect()
+    }
+
+    fn into_map(self) -> BTreeMap<PathBuf, String> {
+        let mut desktop_files = self.system;
+        if env::home_dir() != Some(self.home_dir) {
```

Comment:
I think when DistroShelf is running as a flatpak, `env::home_dir` is different than the $HOME of the host?

By running
```
flatpak enter com.ranfdev.DistroShelf /usr/bin/bash
bash-5.3$ echo $HOME
```
i get an empty output, so this check doesn't seem to work when running DistroShelf as flatpak